app/src/sharedTest/java/org/oppia/app/recyclerview/BindableAdapterTest.kt

```diff
@@ -251,6 +251,62 @@ class BindableAdapterTest {
     }
   }
 
+  @Test
+  fun testBindableAdapter_incomingSameData_noRebindingShouldHappen() {
+
+    val adapter = createMultiViewTypeNoDataBindingBindableAdapter()
+    BindableAdapterTestFragmentPresenter.testBindableAdapter = adapter
+
+    lateinit var dataObserver: RecyclerView.AdapterDataObserver
+    var doesRebind = false
+
+    val oldList = listOf(STR_VALUE_1, STR_VALUE_1, INT_VALUE_1).toMutableList()
+    val newList = listOf(STR_VALUE_1, STR_VALUE_1, INT_VALUE_1).toMutableList()
+
+    ActivityScenario.launch(BindableAdapterTestActivity::class.java).use { scenario ->
+
+      scenario.onActivity { activity ->
+        dataObserver = object : RecyclerView.AdapterDataObserver() {
+          override fun onChanged() {
+            doesRebind = true
+          }
+
+          override fun onItemRangeInserted(positionStart: Int, itemCount: Int) {
+            assertThat(positionStart).isEqualTo(0)
+            assertThat(itemCount).isEqualTo(3)
+          }
+
+          override fun onItemRangeChanged(positionStart: Int, itemCount: Int) {
+            doesRebind = true
+          }
+
+          override fun onItemRangeChanged(positionStart: Int, itemCount: Int, payload: Any?) {
+            doesRebind = true
+          }
+        }
+        adapter.registerAdapterDataObserver(dataObserver)
+      }
+
+      scenario.onActivity { activity ->
+        val liveData = getRecyclerViewListLiveData(activity)
+        liveData.value = oldList
+      }
+      safelyWaitUntilIdle()
```

Comment:
Suggest moving this up to onActivity to a helper method to simplfiy this & the next line.